module/modmanager/manager.go

```diff
@@ -510,6 +535,44 @@ func (mgr *Manager) getModule(conf resource.Config) (*module, bool) {
 	return nil, false
 }
 
+// CleanModuleDataDirectory removes unexpected folders and files from the robot's module data directory.
+func (mgr *Manager) CleanModuleDataDirectory(ctx context.Context) error {
+	if mgr.moduleDataParentDirectory == "" {
+		return errors.New("cannot clean a root level module data directory")
+	}
+	// Absolute path to all dirs that should exist
+	expectedDirs := make(map[string]bool, len(mgr.modules))
+	for _, m := range mgr.modules {
+		expectedDirs[m.dataDir] = true
+	}
+	// If there are no expected directories, we can shortcut and early-exit
+	if len(expectedDirs) == 0 {
+		if err := os.RemoveAll(mgr.moduleDataParentDirectory); err != nil {
```

Comment:
It'd be good to have an info log saying we're removing (the value of) `moduleDataParentDirectory`.

Reply:
Great suggestion.

Reply:
Added.